File: backend/internal/dto/response/exchange.go
package response

import "time"

type ExchangePreviewResp struct {
	FromCurrency   string `json:"from_currency"`
	ToCurrency     string `json:"to_currency"`
	FromAmount     string `json:"from_amount"`
	ToAmount       string `json:"to_amount"`
	ExchangeRate   string `json:"exchange_rate"`
	PlatformFee    string `json:"platform_fee"`
	FeeCurrency    string `json:"fee_currency"`
	NetToAmount    string `json:"net_to_amount"`
	TotalDeduction string `json:"total_deduction"`
}

type ExchangeOrderResp struct {
	ID           uint64    `json:"id"`
	ExchangeType string    `json:"exchange_type"`
	FromCurrency string    `json:"from_currency"`
	ToCurrency   string    `json:"to_currency"`
	FromAmount   string    `json:"from_amount"`
	ToAmount     string    `json:"to_amount"`
	ExchangeRate string    `json:"exchange_rate"`
	PlatformFee  string    `json:"platform_fee"`
	Status       string    `json:"status"`
	FailReason   string    `json:"fail_reason,omitempty"`
	CreatedAt    time.Time `json:"created_at"`
}

type ExchangeOrderListResp struct {
	Orders []ExchangeOrderResp `json:"orders"`
	Total  int64               `json:"total"`
}

File: admin/src/types/exchange.ts
export interface AdminExchange {
  id: number;
  platform_order_id: string;
  merchant_id: number;
  merchant_email: string;
  exchange_type: string;
  from_currency: string;
  to_currency: string;
  from_amount: string;
  to_amount?: string;
  exchange_rate?: string;
  platform_fee: string;
  status: string;
  fail_reason?: string;
  kun_order_id?: string;
  kun_request_no?: string;
  created_at: string;
  completed_at?: string;
}

export interface AdminExchangeListResponse {
  exchanges: AdminExchange[];
  total: number;
}

export interface AdminExchangeSyncResponse {
  order_id: number;
  status: string;
  kun_status: string;
  updated: boolean;
  to_amount?: string;
  exchange_rate?: string;
  platform_fee: string;
  fail_reason?: string;
}
